packages/synapse-react-client/src/components/SynapseTable/datasets/DatasetItemsEditor.tsx

```diff
@@ -113,6 +113,72 @@ export function getCopy(entity?: EntityRefCollectionView) {
   }
 }
 
+function getDataSetDifference(
+  oldDataSet: EntityRef[],
+  changedItems: EntityRef[],
+) {
+  const unchangedItems = oldDataSet.filter(
+    oldItem =>
+      !changedItems.find(newItem => newItem.entityId === oldItem.entityId),
+  )
+  const deletedItems = [...unchangedItems]
+  const { updatedItems, newItems } = changedItems.reduce(
+    (results, result) => {
+      const oldItem = oldDataSet.find(old => old.entityId === result.entityId)
+      if (oldItem) {
+        if (result.versionNumber !== oldItem.versionNumber) {
+          results['updatedItems'].push(result)
+        } else {
+          unchangedItems.push(result)
+        }
+      } else {
+        results['newItems'].push(result)
+      }
+      return results
+    },
+    { updatedItems: [], newItems: [] } as {
+      updatedItems: EntityRef[]
+      newItems: EntityRef[]
+    },
+  )
+
+  return { unchangedItems, updatedItems, newItems, deletedItems }
+}
+
+function getDatasetChangedToastMessageTitle(
```

Comment:
extracted from component, only change is moved external variable references to function arguments